File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        int inversions = 0, n = nums.size();
        for (int i = 0; i < n - 1; i++) {
            if (nums[i] > nums[i + 1]) {
                inversions++;
            }
        }
        if (inversions == 0)
            return true;
        else if (inversions == 1 && nums[n - 1] <= nums[0])
            return true;
        else
            return false;
    }
};

File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var check = function (nums) {
    let inv = 0, n = nums.length;
    for (let i = 0; i < n - 1; i++) {
        if (nums[i] > nums[i + 1])
            inv++;
    }
    if (!inv) return true;
    if (inv == 1 && nums[0] >= nums[n - 1]) return true;
    return false;
};